rest/snapshot_test.go

```diff
@@ -596,3 +598,169 @@ func TestGetIndicesSnapshot(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, &expect, res)
 }
+
+func TestListAssetSnapshots(t *testing.T) {
+	c := polygon.New("API_KEY")
+
+	httpmock.ActivateNonDefault(c.HTTP.GetClient())
+	defer httpmock.DeactivateAndReset()
+
+	expectedResponse := `{
+	 "results": [
+		` + indent(true, assetSnapshotsTestData[0], "\t\t") + `,
+		` + indent(true, assetSnapshotsTestData[1], "\t\t") + `,
+		` + indent(true, assetSnapshotsTestData[2], "\t\t") + `
+	 ],
+	 "status": "OK",
+	 "request_id": "0d350849-a2a8-43c5-8445-9c6f55d371e6",
+	 "next_url": "https://api.polygon.io/v3/snapshot/cursor=YXA9MSZhcz0mbGltaXQ9MSZzb3J0PXRpY2tlcg"
+	}`
+
+	registerResponder("https://api.polygon.io/v3/snapshot?ticker.any_of=AAPL%2CMETA%2CF", expectedResponse)
+	registerResponder("https://api.polygon.io/v3/snapshot/cursor=YXA9MSZhcz0mbGltaXQ9MSZzb3J0PXRpY2tlcg", "{}")
+
+	iter := c.ListAssetSnapshots(
+		context.Background(),
+		models.ListAssetSnapshotsParams{}.WithTickerAnyOf("AAPL,META,F"),
+	)
+
+	// iter creation
+	require.NoError(t, iter.Err())
+	require.NotNil(t, iter.Item())
+
+	// correct values
+	var iterCount int
+	for iter.Next() {
+		var wantSnapshot models.SnapshotResponseModel
+		err := json.Unmarshal([]byte(assetSnapshotsTestData[iterCount]), &wantSnapshot)
+		require.Nil(t, err)
+
+		require.Nil(t, iter.Err())
+		assert.Equal(t, wantSnapshot, iter.Item())
+		iterCount++
+	}
+
+	assert.Equal(t, len(assetSnapshotsTestData), iterCount, fmt.Sprintf("expected %d iterators", len(assetSnapshotsTestData)))
+	assert.False(t, iter.Next())
+	assert.Nil(t, iter.Err())
+}
+
+var assetSnapshotsTestData = []string{
```

Comment:
Since this endpoint is polymorphic, it might make sense to add different types of snapshots to the response test data. right now, it only has stocks snapshots, but you could add an options snapshot, and and error result to make sure those are unmarshaled correctled